ducktape/services/service.py

```diff
@@ -100,11 +102,29 @@ def start(self):
 
         self.logger.debug(self.who_am_i() + ": killing processes and attempting to clean up before starting")
         for node in self.nodes:
-            # Added precaution - kill running processes
-            self._kill_running_processes(node)
-            self.force_clean_node(node)
-            self.stop_node(node)
-            self.clean_node(node)
+            # Added precaution - kill running processes, clean persistent files
+            # try/except for each step, since each of these steps may fail if there are no processes
+            # to kill or no files to remove
+
+            try:
+                self._kill_running_processes(node)
+            except:
+                pass
+
+            try:
+                self.force_clean_node(node)
+            except:
+                pass
+
+            try:
+                self.stop_node(node)
```

Comment:
For now I'd prefer to remove these completely since they don't make a lot of sense in the more general setting, and could be somewhat dangerous (reckless removal of files, reckless destruction of java processes) if used in the wrong context. 

Miscellaneous thoughts regarding cleanup:
Default "nuclear cleanup" behavior should probably take into account some state associated with cluster type. 

For file cleanup, maybe the cluster specifies a root directory where any logs/persistent state goes by convention. 

For process cleanup, we need a more uniform approach to process handling. Maybe services register one or more pids when they start (e.g. a mapping { key => pid } where key is something like node + extra info (extra info is needed for services such as Hadoop that start multiple processes per node)). Where these pids are persisted would depend on the cluster type.